### PR TITLE
Distinct execution

### PR DESCRIPTION
**Pull request by narekhovhannisyan**
### Types of changes

- [ ] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

- 
-

### CODE REVIEW

File: src/if-run/lib/compute.ts [hunk 1]
```diff
@@ -48,12 +50,13 @@ const mergeDefaults = (
  * 4. Goes through the pipeline plugins, by checking if it's `execute` plugin. If so sets outputs.
  *    If is a `groupby` plugin, it will return child components rather than outputs.
```
Comment (manushak) [inline src/if-run/lib/compute.ts:51]: update the comment here

File: src/if-run/lib/regroup.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+const {INVALID_GROUP_KEY, REGROUP_ERROR} = STRINGS;
+
+/**
+ * Grouping strategy.
+ */
+export const Regroup = (inputs: PluginParams[], groups: string[]) => {
+  /**
+   * Creates structure to insert inputs by groups.
+   */
+  const appendGroup = (
```
Comment (manushak) [inline src/if-run/lib/regroup.ts:20]: create some options property with custom type, or at least define the `object` property type and remove the `any` return type

File: src/if-run/lib/regroup.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      }
+
+      appendGroup(value, object.children[group], groups);
+    }
+
+    return object;
+  };
+
+  /**
+   * Validates config parameter.
```
Comment (manushak) [inline src/if-run/lib/regroup.ts:49]: update comment

**Review by jmcook1186 — APPROVED** [review]
Seems to be working as expected - nice work!

**Review by manushak — APPROVED** [review]

**Pull request merged**